### DISCUSSION TITLE
Don't bump Python version

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [renovatebot/renovate#30770] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, 37.440.7

### Please tell us more about your question or problem

- For tests, a job in my `.gitlab-ci.yml` uses the `python:3.11` image.
- This version should not be bumped.
- Renovate is bumping the version to `python:3.12`.

Using constraints doesn't help. I suppose they relate to dependencies only.

```
"constraints": {
    "python": "3.11"
}
```

Are package rules the way to go? Something like:

```
{
  "packageRules": [
    {
      "matchPackageNames": ["python"],
      "allowedVersions": "3.11"
    }
  ]
}
```

Or am I missing something trivial?

**Comment by rarkins** [comment, marked as answer]
Yes, use allowedVersions. Or maybe dependencyDashboardApproval so that one day when you want to upgrade your 3.12 then you can request it

**Reply by WilliamDEdwards** [reply]
Useful! Thanks for the hint.

**Comment by HonkingGoose** [comment]
> GitLab, 37.440.7

By the way you should update your version of Renovate to `v38`. :wink:
